projecteuler/helpers/factors.py

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env python3
+
+def get_factors(number, list_to_filter_by=[]):
```

Comment:
this is the "default list" thing I mentioned. Try this:

```python
def horrible_append(value, initial=[]): 
    initial.append(value) 
    return initial 

print(horrible_append("hello", []))
print(horrible_append("world", []))  
print(horrible_append("world", ["hello"]))
print(horrible_append("hello"))
print(horrible_append("world"))                  
```

output:

```
['hello']
['world']
['hello','world']
['hello']
['hello','world']
```

Reply:
out of interest, why does it have a filtered list? only its own unit test actually uses it

Reply:
there is a much quicker way to do this by only testing up to `sqrt(number)` and using the result of the division

Reply:
> out of interest, why does it have a filtered list? only its own unit test actually uses it

Haha! I think I was using it and then changed the method that called it, so didn't need it but left it with the tests. I'll remove it. That last append is crazy, would no way have realised that!

Reply:
After a bit of playing around, think I've sped it up significantly thanks to the `sqrt` pointer 👍